gravity.sh

```diff
@@ -768,25 +768,21 @@ gravity_ParseFileIntoDomains() {
   tr '[:upper:]' '[:lower:]' < "${src}" > "${destination}"
 
   # 2) Remove carriage returns
-  sed -i 's/\r$//' "${destination}"
-
-  # 3a) Remove comments (text starting with "#", include possible spaces before the hash sign)
-  sed -i 's/\s*#.*//g' "${destination}"
-
-  # 3b) Remove lines starting with ! (ABP Comments)
-  sed -i 's/\s*!.*//g' "${destination}"
-
-  # 3c) Remove lines starting with [ (ABP Header)
-  sed -i 's/\s*\[.*//g' "${destination}"
-
-  # 4) Remove lines containing "/"
-  sed -i -r '/(\/).*$/d' "${destination}"
-
-  # 5) Remove leading tabs, spaces, etc. (Also removes leading IP addresses)
-  sed -i -r 's/^.*\s+//g' "${destination}"
-
-  # 6) Remove empty lines
-  sed -i '/^$/d' "${destination}"
+  # 3) Remove lines starting with ! (ABP Comments)
+  # 4) Remove lines starting with [ (ABP Header)
+  # 5) Remove lines containing ABP extended CSS selectors ("##", "#!#", "#@#", "#?#")
+  # 6) Remove comments (text starting with "#", include possible spaces before the hash sign)
+  # 7) Remove leading tabs, spaces, etc. (Also removes leading IP addresses)
+  # 8) Remove empty lines
+
+    sed -i -r \
+    -e 's/\r$//' \
+    -e 's/\s*!.*//g' \
+    -e 's/\s*\[.*//g' \
+    -e '/\#[!|?|@]{0,1}\#/d' \
```

Comment:
1. You don't need the `|` here. This `[!?@]` already means "Match a single character present in the list".
2. `#$#` is also a possible rule. You forgot to include the `$` to the regex.
3. **Note:** I'm not sure if there is a `#!#` rule.

Reply:
You're 100% correct.